docs/component-guides/KServe.md

```diff
@@ -0,0 +1,181 @@
+1. `export certArn=<>` 
+
+If you use Dex as auth provider in your Kubeflow deployment, refer the [Dex specific section](#dex---update-the-load-balancer). If you use Cognito, continue to the next section
+
+## Cognito - Create a HTTP header evaluation based Load Balancer
+
+Currently, it is not possible to programatically authenticate a request through ALB that is using Amazon Cognito to authenticate users. (i.e. you cannot generate the `AWSELBAuthSessionCookie` cookies by using the access tokens from Cognito). Instead of using the existing ALB, we will be creating another Application Load Balancer(ALB) endpoint which authorizes based on custom strings specified in a predefined HTTP header. We will use an ingress to set [HTTP header conditions](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-listeners.html#http-header-conditions) on ALB which will configure rules that route requests based on the HTTP headers in the request. This can be used for service to service communication in your application.
+
+### Create Ingress
+
+1. Configure the following parameters for [ingress](../../awsconfigs/common/istio-ingress/overlays/api/params.env):
+    - certArn: ARN of certificate created in previous step
+    - (optional) httpHeaderName: Custom HTTP header name you want to configure for the rule evaluation. Default is `x-api-key`
+    - httpHeaderValues: one or more match strings that need to be compared against the header value in the request received. You only need to pass one of the tokens in the request. Pick strong values.
+    1. Replace the `token1` string with a token of your choice and optionally the httpHeaderName
+        ```
+        printf '
+        certArn='$certArn'
+        httpHeaderName=x-api-key
+        httpHeaderValues=["token1"]
+        ' > awsconfigs/common/istio-ingress/overlays/api/params.env
+        ```
+1. Create the ingress by running the following command:
+    1. ```
+       kustomize build awsconfigs/common/istio-ingress/overlays/api | kubectl apply -f -
+       ```
+1. Check if the ingress managed ALB is provisioned. It takes around 3-5 minutes
+    1. ```
+        kubectl get ingress -n istio-system istio-ingress-api
+        NAME                CLASS    HOSTS   ADDRESS                                                                PORTS   AGE
+        istio-ingress-api   <none>   *       xxxxxx-istiosystem-istio-2af2-1100502020.us-west-2.elb.amazonaws.com   80      14m
+        ```
+1. Once ALB is ready, you will need to add a DNS record for the staging subdomain. [Add DNS records for the staging subdomain with ALB address](#add-dns-records-for-the-staging-subdomain-with-alb-address) section will guide you through it
+
+## Dex - Update the Load Balancer
+
```

Comment:
I think below part is common in cognito as well. 
shall we just omit that saying you don't need to do anything for dex if you've followed the [Exposing Kubeflow over Load Balancer guide](../deployment/add-ons/load-balancer/README.md) for non-Cognito based deployment option e.g. [Vanilla](../deployment/vanilla/README.md)(uses Dex as auth provider). ?

Reply:
we are updating different manifests in dex vs cognito and name of ingress resource is different(istio-ingress vs istio-ingress-api). did I understand your comment correctly?